Update dependencies and add Nix flake

willcohen opened the pull request:
I'm working on dusting off my branch for #69, so in advance of that, I thought it might be worth tidying up the aging dependency list on the main branch first!

willcohen committed:
Update dependencies

willcohen commented:
This also should close #74, though not till it's incorporated into a release

isaacbrodsky reviewed (approved):
h3-java update looks good to me

willcohen committed:
Add nix flake and clj-build framework for CI

willcohen commented:
In addition, I just added a little more housekeeping, to try to prepare for adding CI back in (using GH actions, maybe? The cljs branch definitely will need it, since there's more targets to check!). I added a basic Nix configuration so there's a reproducible environment that can be generated, and I added `clj-build` with a goal of starting to move builds toward `deps.edn`, now that Clojure's got `tools.build`.

willcohen committed:
Add build.clj

worace commented:
@willcohen great to see you are messing around with this again! I am afraid I haven't touched it in a while.

RE:

> to prepare for adding CI back in (using GH actions, maybe?)

Can you remind me where we left off on that? I guess we took the old CI off because travis shut down and we never got a new one set up?

willcohen commented:
I don't think we ever discussed it! I think what you said is exactly the current state of it. GitHub Actions seems like they work fine for lots of other projects. The main blocker for me and the clojurescript branch is the fact that I have to rewrite the entire test suite since midje isn't cross compatible, so it's a good chance to rethink through CI since the test suite could see a big change soon too. `tools.build` seems like a pretty solid official-ish successor to lots of the things that `geo` still depends on `lein` for, so if it's okay I might try to think through a CI process that lines up with `clj`/`tools.build`/`clojure.test` and perhaps GH actions, and we can kick the tires on some options once I get that a little further along?

willcohen committed:
Update dependencies
